NuRadioReco/detector/generic_detector.py

```diff
@@ -51,46 +51,67 @@ def __init__(self, json_filename, default_station, default_channel=None, assume_
         assume_inf : Bool
             Default to True, if true forces antenna madels to have infinite boundary conditions, otherwise the antenna madel will be determined by the station geometry.
         """
-        super(GenericDetector, self).__init__('json', json_filename, assume_inf)
+        super(GenericDetector, self).__init__(source, json_filename, dictionary, assume_inf)
+
         self.__default_station_id = default_station
+        self.__default_channel_id = default_channel
+        self.__station_changes_for_event = []
+        self.__run_number = None
+        self.__event_id = None
         if not self.has_station(self.__default_station_id):
             raise ValueError('The default station {} was not found in the detector description'.format(self.__default_station_id))
+
         Station = Query()
         self.__default_station = self._stations.get((Station.station_id == self.__default_station_id))
+
         if default_channel is not None:
             Channel = Query()
             self.__default_channel = self._channels.get((Channel.station_id == default_station)&(Channel.channel_id == default_channel))
             if self.__default_channel is None:
                 raise ValueError('The default channel {} of station {} was not found in the detector description'.format(default_channel, self.__default_station_id))
         else:
             self.__default_channel = None
-    def _query_station(self, station_id):
+
+    def _get_station(self, station_id):
+        if(station_id not in self._buffered_stations.keys()):
+            self._buffer(station_id)
+        res = copy.copy(self._buffered_stations[station_id])
+        if self.__run_number is not None and self.__event_id is not None:
+            for change in self.__station_changes_for_event:
+                if change['station_id'] == station_id and change['run_number'] == self.__run_number and change['event_id'] == self.__event_id:
+                    for name, value in change['properties'].items():
+                        res[name] = value
```

Comment:
This implementation is slightly different than what I recall we discussed given that I understand it correctly. 

I thought changes to the detector description would be saved incrementally with each event. E.g. event 3 changes the antenna 1 to bicone, and event 6 changes antenna 2 to LPDA. Then event 7 would have both changes. 

seems to update a generic detector (the same for all events) with requested changes of this event. So the next event, it not containing its own overwrites, just return the generic detector. Or am I missing something? 

In any case, we should add more documentation to the code, maybe at the header of this file .

Reply:
You are right, each change only applies to one event. I think doing it incrementally could cause problems if events are removed from the file. The detector is only saved for the events that are actually in the file. So in your example, if we applied some quality cuts and got rid of event 6, we wouldn't know that antenna 2 is an LPDA in event 7.
There are probably ways around this, but it seems pretty messy and error-prone, so I decided to store changes per event.

Reply:
Sounds good, I actually prefer this implementation. So far this anyway only applies to coreas simulations where each event has a different station configuration.